src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinder.cs

```diff
@@ -760,7 +760,7 @@ private Expr CreateArgumentEXPR(ArgumentObject argument, LocalVariableSymbol loc
                 flags, name, typeArgumentsAsTypeArray, kind, callingType, null, null, new CMemberLookupResults(
                     _semanticChecker.getBSymmgr().AllocParams(callingTypes.Count, callingTypes.ToArray()),
                     name));
-            if (callingObject.isCLASS())
+            if (callingObject is ExprClass)
```

Comment:
The only two cases that aren't sealed are `ExprLocal` and `ExprStatement` but the only time we check them with `is` it's the new C#7 syntax so it's what would have been an `as` followed by a null-check if using C#6 and since the `as` cast is going to happen for some paths any way, it's not as useful to have another way of checking the type.